File: app/src/main/java/uiapp/ui/ktx/KtxViewModel.java
package uiapp.ui.ktx;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;


public class KtxViewModel extends ViewModel {
    private final SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS", Locale.US);

    public MutableLiveData<String> date = new MutableLiveData<>(format.format(new Date()));

}

File: app/src/main/java/uiapp/ui/ktx/KtxViewModel.kt
package uiapp.ui.ktx

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

class KtxViewModel : ViewModel() {
    private val format = SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS", Locale.US)

    val date: MutableLiveData<String> = MutableLiveData(format.format(Date()))
}
